model-sync-gradle/src/main/kotlin/org/modelix/model/sync/gradle/ModelSyncGradlePlugin.kt

```diff
@@ -0,0 +1,220 @@
+package org.modelix.model.sync.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.Sync
+import org.gradle.api.tasks.TaskProvider
+import org.modelix.model.sync.gradle.config.LocalSource
+import org.modelix.model.sync.gradle.config.LocalTarget
+import org.modelix.model.sync.gradle.config.ModelSyncGradleSettings
+import org.modelix.model.sync.gradle.config.ServerSource
+import org.modelix.model.sync.gradle.config.ServerTarget
+import org.modelix.model.sync.gradle.config.SyncDirection
+import org.modelix.model.sync.gradle.tasks.ExportFromModelServer
+import org.modelix.model.sync.gradle.tasks.ExportFromMps
+import org.modelix.model.sync.gradle.tasks.GenerateAntScriptForMps
+import org.modelix.model.sync.gradle.tasks.ImportIntoModelServer
+import org.modelix.model.sync.gradle.tasks.ImportIntoMps
+import org.modelix.model.sync.gradle.tasks.ValidateSyncSettings
+import java.io.File
+import java.net.URL
+import java.util.Enumeration
+import java.util.Properties
+
+class ModelSyncGradlePlugin : Plugin<Project> {
+
+    private lateinit var settings: ModelSyncGradleSettings
+
+    override fun apply(project: Project) {
+        settings = project.extensions.create("modelSync", ModelSyncGradleSettings::class.java)
+        getBaseDir(project).mkdirs()
+
+        project.afterEvaluate {
+            val validateSyncSettings = project.tasks.register("validateSyncSettings", ValidateSyncSettings::class.java) {
+                settings.taskDependencies.forEach { dependency ->
+                    it.dependsOn(dependency)
+                }
+                it.settings.set(settings)
+            }
```

Comment:
Why is this validation done that late? Wouldn't it be possible to immediately fail the build in case things are configured improperly?